src/test/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParserV2gTest.java

```diff
@@ -0,0 +1,415 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.mad.parsers;
+
+import com.arpnetworking.metrics.common.parsers.exceptions.ParsingException;
+import com.arpnetworking.metrics.mad.model.Metric;
+import com.arpnetworking.metrics.mad.model.Record;
+import com.arpnetworking.tsdcore.model.Key;
+import com.arpnetworking.tsdcore.model.Quantity;
+import com.arpnetworking.tsdcore.model.Unit;
+import com.google.common.io.Resources;
+import org.hamcrest.Matchers;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Tests for the 2f version of the query log format.
+ *
+ * @author Ryan Ascheman (rascheman at groupon dot com)
+ */
+public class JsonToRecordParserV2gTest {
```

Comment:
Missing a test that actually tests that dimensions present in the JSON are found on the parsed record.